### PR TITLE
added handling for boolean in usertokens with unit tests

### PR DESCRIPTION
**Pull request by brianonn**
#### Description
fixes https://github.com/aporeto-inc/aporeto/issues/1655

### CODE REVIEW

**Comment by mheese** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1311",
    "commit-sha": "ea05515051005de64332042c66d47e08f6c3ebb8"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "e8c35bed3e00b2f4d07bf076d5b7cb4ff98d0507"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1311",
    "commit-sha": "ea05515051005de64332042c66d47e08f6c3ebb8"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "c21d80064b30f57834408d80c74491fa7d5ea8ec"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1311",
    "commit-sha": "ea05515051005de64332042c66d47e08f6c3ebb8"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "70e6e20c959b821ba145dd627fb580653ecfdf5a"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1311",
    "commit-sha": "ea05515051005de64332042c66d47e08f6c3ebb8"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "2a0588f72f5714e14e469f1ff634e839e57c2ccb"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1311",
    "commit-sha": "ea05515051005de64332042c66d47e08f6c3ebb8"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "1175eb25d4b685477ee3243746892f547847e1d2"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1311",
    "commit-sha": "ea05515051005de64332042c66d47e08f6c3ebb8"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "0474b1ee3582767f7936b44349a9b540b5aa32aa"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1311",
    "commit-sha": "ea05515051005de64332042c66d47e08f6c3ebb8"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "493a0e1afb450fbbfbb3f50c000627066c985648"
  }
]
```

File: controller/pkg/usertokens/common/common.go [hunk 1]
```diff
@@ -9,21 +14,67 @@ const (
...
+	case uint, uint8, uint16, uint32, uint64:
+		attributes = append(attributes, key+"="+strconv.FormatUint(toUint64(claim), 10))
+	case float32:
+		attributes = append(attributes, key+"="+strconv.FormatFloat(float64(claim), 'G', -1, 32))
+	case float64:
+		attributes = append(attributes, key+"="+strconv.FormatFloat(claim, 'G', -1, 64))
+	case string:
+		attributes = append(attributes, key+"="+claim)
+	case []string:
+		for _, data := range claim {
```
Comment (dstiliadis) [inline controller/pkg/usertokens/common/common.go:71]: as discussed, a slice will most likely unmarshall to []interface{} .. so most likely the best way is call again the flattenclaims but with the slice array. This is the reason that []string itself doesn't work some times.

Comment (brianonn) [reply controller/pkg/usertokens/common/common.go:71]: This commit addresses the case when []interface{} :  7308a498

**Comment by brianonn** [comment]
https://ci.aporeto.io/teams/main/pipelines/pr-tests/jobs/functional-tests/builds/149

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "d46852c91f63d6e7d4d27ab0134d9e2fa4b778a2"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "f05416e4c6e031d0d97ca9f181ddb2da481d5091"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "f13c11353a6e3026cf87be19ad1697170cb2c363"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "da97f0397f6cd222811d10c5c33c5233ab9d07b0"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "f13c11353a6e3026cf87be19ad1697170cb2c363"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "da97f0397f6cd222811d10c5c33c5233ab9d07b0"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "f13c11353a6e3026cf87be19ad1697170cb2c363"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "da97f0397f6cd222811d10c5c33c5233ab9d07b0"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "9ca6db4cd4997a2dbb5e6bd75317e7a21781ec37"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "da97f0397f6cd222811d10c5c33c5233ab9d07b0"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "9ca6db4cd4997a2dbb5e6bd75317e7a21781ec37"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "da97f0397f6cd222811d10c5c33c5233ab9d07b0"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "9ca6db4cd4997a2dbb5e6bd75317e7a21781ec37"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "da97f0397f6cd222811d10c5c33c5233ab9d07b0"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "9ca6db4cd4997a2dbb5e6bd75317e7a21781ec37"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "3646aeda9c4727fbfde08907dcb739458b1685ed"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "3646aeda9c4727fbfde08907dcb739458b1685ed"
  },
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "0e5d143d9831c39323352b0d798f74b652668659"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "c8d78e753d9a7154d13bea7d7cba858fd9c8574c"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "c8d78e753d9a7154d13bea7d7cba858fd9c8574c"
  },
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "4cb32cd11744afe54dd6730ed79396bc32db2529"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "ae53f289a599a6c21c93fe12c54c44a74e3bc19f"
  },
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "1f5f98e8770d8598bdded8a68cc4422f33627e04"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "4a8953eea0b24855dd8e93e0179e76a159560887"
  },
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "7e7893fca220eb183817e38d7d6eb5ecd6794b83"
  }
]
```

**Comment by brianonn** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "fix-issue-1655",
    "component": "integrations-lib",
    "pr-id": "152",
    "commit-sha": "1caa2139aaa59d0b19611e17e8bfab8ef5d05d54"
  },
  {
    "project": "fix-issue-1655",
    "component": "trireme-lib",
    "pr-id": "843",
    "commit-sha": "4a8953eea0b24855dd8e93e0179e76a159560887"
  },
  {
    "project": "fix-issue-1655",
    "component": "enforcerd",
    "pr-id": "1355",
    "commit-sha": "7e7893fca220eb183817e38d7d6eb5ecd6794b83"
  }
]
```

**Review by mheese — APPROVED** [review]

**Pull request merged**